FileShareClient/FileShareApiClient.cs

```diff
@@ -43,8 +48,8 @@ public async Task<BatchStatusResponse> GetBatchStatusAsync(string batchId)
 
             using (var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, uri))
             {
-                var response = await httpClientFactory.CreateClient()
-                    .SendAsync(httpRequestMessage, CancellationToken.None);
+                var httpClient = await httpClientFactory.CreateClient().SetAuthenticationHeader(authTokenProvider);
```

Comment:
I notice you had to make the SetAuthenticationHeader() call in many places in both the client and admin client. Perhaps it's worth adding a factory method for creating HttpClient instances.

Reply:
We did consider it to put in `HttpClientFactory`'s `CreateClient()` method. However, it is a synchronous method where as getting a token requires an asynchronous method call. Hence we decided to call `SetAuthenticationHeader()` method explicitly in each method that makes http call.